src/monitor-api-requests.ts

```diff
@@ -223,18 +211,18 @@ export async function getRecommendedMonitors(
       templateVariables: recommendedMonitorParam.attributes.template_variables,
     };
 
-    const recommendedMonitorApiId = recommendedMonitorParam.id;
-    if (isValidRecommendedMonitorApiId(recommendedMonitorApiId)) {
-      const recommendedMonitorId = recommendedMonitorApiIdToId[recommendedMonitorApiId];
-      recommendedMonitors[recommendedMonitorId] = recommendedMonitor;
-    }
+    const recommendedMonitorId = parseRecommendedMonitorServerlessId(recommendedMonitorParam);
+    recommendedMonitors[recommendedMonitorId] = recommendedMonitor;
   });
 
   return recommendedMonitors;
 }
 
-function isValidRecommendedMonitorApiId(
-  recommendedMonitorApiId: string,
-): recommendedMonitorApiId is keyof typeof recommendedMonitorApiIdToId {
-  return recommendedMonitorApiId in recommendedMonitorApiIdToId;
+export function parseRecommendedMonitorServerlessId(recommendedMonitorParams: RecommendedMonitorParams): string {
+  for (const tag of recommendedMonitorParams.attributes.tags) {
+    if (tag.startsWith("serverless_id:")) {
+      return tag.substring(tag.indexOf(":") + 1);
+    }
+  }
+  throw new Error(`Recommended monitor ${recommendedMonitorParams.id} does not have a serverless_id tag.`);
```

Comment:
We should not be throwing an error here. The way it should be handled is, in the list of recommended monitors that get returned, we should look for `serverless_id` tag. If it is present, then we should return that template as a monitor that can be configured from plugin. If not, we should skip that template. Here's the reason behind it.

1. **This is not a hard requirement**: From the plugin, we're looking for all serverless monitor templates and then trying to see if there is a template that matches an `id` that user has specified. We could potentially create a monitor template that is meant to be only used from the "Browse Monitor Templates" view. In that case, it would be a lambda serverless monitor template that does not have a `serverless_id` tag.
2. **This check will cause all plugin deploys to fail if we make a mistake in our repo**: If we add a new lambda serverless monitor template that doesn't have `serverless_id` tag in it, all plugin deploys irrespective of whether the customer is using that specific monitor or not will fail because while parsing this error will get thrown.

Reply:
Good point. Let me change it.

Reply:
Updated and retested!